lib/util/source-code.js

```diff
@@ -87,19 +87,28 @@ function looksLikeExport(astNode) {
 
 /**
  * Represents parsed source code.
- * @param {string} text The source code text.
- * @param {ASTNode} ast The Program node of the AST representing the code.
+ * @param {string} text - The source code text. The BOM of this text should be stripped.
+ * @param {ASTNode} ast - The Program node of the AST representing the code.
  * @constructor
  */
 function SourceCode(text, ast) {
-
     validate(ast);
 
+    // Check Unicode BOM.
+    // In JavaScript, string data is stored as UTF-16, so BOM is 0xFEFF.
+    // http://www.ecma-international.org/ecma-262/6.0/#sec-unicode-format-control-characters
+    /**
+     * The flag to indicate that the source code has Unicode BOM.
+     * @type boolean
+     */
+    this.hasBOM = (text.charCodeAt(0) === 0xFEFF);
+
     /**
      * The original text source code.
+     * BOM was stripped from this text.
      * @type string
      */
-    this.text = text;
+    this.text = (this.hasBOM ? text.slice(1) : text);
```

Comment:
But AST is created from the stripped text.
BOM is zero width character but 1 character.
So I guessed AST gets wrong `range` if this is original text.